store_test.go

```diff
@@ -0,0 +1,338 @@
+		date:          date{"2015-10-07T15:00:00.000Z", "2015-10-11T15:00:00.000Z"},
+	}
+
+	return generateMongoQuery(qParams)
+}
+
+func TestStore_generateMongoQuery_basic(t *testing.T) {
+
+	time.Now()
+	query := basicQuery()
+
+	expectedQuery := bson.M{"_groupId": "123",
+		"_active":        true,
+		"_schemaVersion": bson.M{"$gte": 0, "$lte": 2}}
+
+	eq := reflect.DeepEqual(query, expectedQuery)
+	if !eq {
+		t.Error(getErrString(query, expectedQuery))
+	}
+
+}
+
+func TestStore_generateMongoQuery_allparams(t *testing.T) {
+
+	query := allParamsQuery()
+
+	expectedQuery := bson.M{
+		"_groupId":       "123ggf",
+		"_active":        true,
+		"_schemaVersion": bson.M{"$gte": 0, "$lte": 2},
+		"type":           bson.M{"$in": strings.Split("smbg,cbg", ",")},
+		"subType":        bson.M{"$in": strings.Split("stuff", ",")},
+		"time": bson.M{
+			"$gte": "2015-10-07T15:00:00.000Z",
+			"$lte": "2015-10-11T15:00:00.000Z"},
+	}
+
+	eq := reflect.DeepEqual(query, expectedQuery)
+	if !eq {
+		t.Error(getErrString(query, expectedQuery))
+	}
+}
+
+func TestStore_generateMongoQuery_noDates(t *testing.T) {
+
+	query := typeAndSubtypeQuery()
+
+	expectedQuery := bson.M{
+		"_groupId":       "123ggf",
+		"_active":        true,
+		"type":           bson.M{"$in": strings.Split("smbg,cbg", ",")},
+		"subType":        bson.M{"$in": strings.Split("stuff", ",")},
+		"_schemaVersion": bson.M{"$gte": 0, "$lte": 2}}
+
+	eq := reflect.DeepEqual(query, expectedQuery)
+	if !eq {
+		t.Error(getErrString(query, expectedQuery))
+	}
+}
+
+func TestStore_Ping(t *testing.T) {
+
+	store := before(t)
+	err := store.Ping()
+
+	if err != nil {
+		t.Error("there should be no error but got", err.Error())
+	}
+}
+
+func TestStore_IndexUse_basicQuery(t *testing.T) {
+
+	const expectedCursor = "BtreeCursor _groupId_1__active_1__schemaVersion_1"
+
+	store := before(t)
+	sCopy := store.session
+	defer sCopy.Close()
+
+	query := basicQuery()
+
+	var executedPlan map[string]interface{}
+
+	err := mgoDataCollection(sCopy).Find(query).Explain(&executedPlan)
+
+	if err != nil {
+		t.Error("there should be no error execting the query", err.Error())
+	}
+
+	usedCursors := getCursors(executedPlan["allPlans"])
+
+	if contains(usedCursors, baseCursor) == false {
+		t.Errorf("excpected [%s] to be in  [%v]", baseCursor, usedCursors)
+	}
+
+	if contains(usedCursors, typeCursor) {
+		t.Errorf("didn't expect [%s] to be in  [%v]", typeCursor, usedCursors)
+	}
+
+}
+
+func TestStore_IndexUse_fullQuery(t *testing.T) {
+
+	const expectedCursor = "BtreeCursor type_1"
+
+	store := before(t)
+	sCopy := store.session
+	defer sCopy.Close()
+
+	query := allParamsQuery()
+
+	var executedPlan map[string]interface{}
+
+	err := mgoDataCollection(sCopy).Find(query).Explain(&executedPlan)
+
+	if err != nil {
+		t.Error("there should be no error execting the query", err.Error())
+	}
+
+	usedCursors := getCursors(executedPlan["allPlans"])
+
+	if contains(usedCursors, typeCursor) == false {
```

Comment:
IMHO, we should never test against true or false. It should just be `if !contains(userCursors, typeCursor) {`